monix-reactive/shared/src/main/scala/monix/reactive/Consumer.scala

```diff
@@ -369,6 +394,74 @@ object Consumer {
     }
   }
 
+  /** Implementation for [[Consumer.partitionParallel]] */
+  private final class PartitionParallelConsumer[A](nrPartitions: Int, overflowStrategy: OverflowStrategy.Synchronous[A], partition: A => Int, consumer: PartitionedConsumer[A]) extends Consumer[A, Unit] {
+    override def createSubscriber(cb: Callback[Unit], s: Scheduler): (Subscriber[A], AssignableCancelable) = {
+
+      val errors = scala.collection.mutable.ArrayBuffer.empty[Throwable]
+
+      val obs = PublishSubject[A]()
+      val pub = obs.whileBusyBuffer(overflowStrategy)
+      val isUpstreamDone = AtomicBoolean(false)
+
+      val refCount = RefCountCancelable { () =>
+        errors.synchronized {
+          if (errors.nonEmpty) {
+            cb.onError(CompositeException(errors))
+          } else {
+            cb.onSuccess(())
+          }
+        }
+      }
+
+      def subscriber(part: Int): Cancelable = {
+
+        def consumerSubscriber(refID: Cancelable) = new Subscriber[Seq[A]] {
+          implicit def scheduler: Scheduler = s
+
+          def onNext(elements: Seq[A]) =
+            consumer.consume(part, elements).coeval.value match {
```

Comment:
Good point. Another question arises; right now I have this:

```scala
try consumer(part, elements).coeval.value match {
  case Left(future) =>
    future.map(_ => Continue)
  case Right(()) =>
    Continue
} catch {
  case NonFatal(ex) =>
    onError(ex)
    Stop
}
```

The part `future.map(_ => Continue)` only deals with the happy path. Should we also deal with errors?

Adding a `.recover` case like this:

```scala
.recover {
  case NonFatal(ex) =>
    onError(ex)
    Stop
}
```